### PR TITLE
Have a better error message for the classic case of a broken tree of MGs

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #23641
 Already tested in tests/mgs/final_mg

### CODE REVIEW

File: framework/src/base/MeshGeneratorSystem.C [hunk 1]
```diff
@@ -259,9 +259,13 @@ MeshGeneratorSystem::createMeshGeneratorOrder()
 
         oss << gen->name();
       }
-      oss << "\"\n\nThis may be due to a missing dependency or may be intentional. Please "
-             "select the final MeshGenerator in\nthe [Mesh] block with the \"final_generator\" "
-             "parameter or add additional dependencies to remove the ambiguity.";
+      oss << "\"\n\nThis may be due to a missing dependency or may be intentional. Please either\n"
+             "- check that all the mesh generators are connected as a tree and culminate in a "
+             "single final mesh. Having one wrong 'input=mg' parameter is the most common error \n"
```
Comment (loganharbour) [inline framework/src/base/MeshGeneratorSystem.C:264]:
```suggestion
             "single final mesh. Having one wrong 'input=mg' parameter is the most common error\n"
```

File: framework/src/base/MeshGeneratorSystem.C [hunk 2]
```diff
@@ -259,9 +259,13 @@ MeshGeneratorSystem::createMeshGeneratorOrder()
...
+             "- add additional dependencies to remove the ambiguity if you are using a user-built "
+             "MeshGenerator\n"
+             "- if you intend to only part of the generators defined (uncommon), select the "
```
Comment (loganharbour) [inline framework/src/base/MeshGeneratorSystem.C:267]:
```suggestion
             "- if you intend to execute a subset of the defined generators (uncommon), select the "
```

**Review by loganharbour — CHANGES_REQUESTED** [review]

File: framework/src/base/MeshGeneratorSystem.C [hunk 3]
```diff
@@ -259,9 +259,13 @@ MeshGeneratorSystem::createMeshGeneratorOrder()
...
       }
-      oss << "\"\n\nThis may be due to a missing dependency or may be intentional. Please "
-             "select the final MeshGenerator in\nthe [Mesh] block with the \"final_generator\" "
-             "parameter or add additional dependencies to remove the ambiguity.";
+      oss << "\"\n\nThis may be due to a missing dependency or may be intentional. Please either\n"
+             "- check that all the mesh generators are connected as a tree and culminate in a "
+             "single final mesh. Having one wrong 'input=mg' parameter is the most common error\n"
+             "- add additional dependencies to remove the ambiguity if you are using a user-built "
+             "MeshGenerator\n"
+             "- if you intend to execute a subset of the defined generators (uncommon), select the "
```
Comment (GiudGiud) [inline framework/src/base/MeshGeneratorSystem.C:267]:
```suggestion
             "- if you intend to execute a subset of the defined generators (uncommon), select the"
```

**Comment by GiudGiud** [comment]
Test failure is unrelated

**Review by loganharbour — APPROVED** [review]

**Pull request merged**